system/admin/filesystem.go

```diff
@@ -1,10 +1,41 @@
 package admin
 
 import (
+	"encoding/json"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ponzu-cms/ponzu/system/db"
+	"github.com/ponzu-cms/ponzu/system/item"
 )
 
+func deleteUploadFromDisk(target string) error {
+	// get data on file
+	data, err := db.Upload(target)
+	if err != nil {
+		return err
+	}
+
+	// unmarshal data
+	upload := item.FileUpload{}
+	if err = json.Unmarshal(data, &upload); err != nil {
+		return err
+	}
+
+	// split and rebuild path in OS friendly way
+	// use path to delete the physical file from disk
+	pathSplit := strings.Split(upload.Path, "/")
+	pathJoin := filepath.Join(pathSplit[2:]...)
```

Comment:
nice!

another nit, but may want to do a bounds check so we return an error vs. panic if there is some case that the path isn't correct/malformed compared to our expectations.

Reply:
Not sure I follow. All errors are checked and returned?  And ```upload.Path``` can only come from the DB. If it's malformed we have problems elsewhere? Could you expand on what you mean here - I'm probably missing something.

Reply:
Ah, sorry, I mean checking the length of `pathSplit` to be `>= 2` (?) so that the operation `pathSplit[2:]` doesn't panic. 

You're right, we would have problems elsewhere for sure :) but it's just become habitual to check for these so panics don't arise. In the case that `pathSplit` length does not meet the criteria, we could return an error.

Reply:
@olliephillips  Actually, we might want to just process the `upload.Path` so as to `strings.TrimPrefix(upload.Path, "/api/")`, or whatever the prefix is, before splitting and re-joining. Then we shouldn't need to mess with the re-slice at all.

Reply:
Got it. That's experience there @nilslice - I didn't see that panic opportunity :) 
I'll revise per your second comment, so we won't need to reslice and can pass the whole slice to ```filepath.Join(...)``` with no panic risk. 👍